### PR TITLE
Detect if watchman is installed

### LINKED ISSUE #22: Detect missing watchman
The extension relies on watchman right? Then we should maybe add a mechanism to detect if the user doesn't have watchman installed and warn them.

### PR DESCRIPTION
**Pull request by mellson**
I've tried to add detection of whether or not watchman is installed on the users system. I can verify that it works as intended on macOS, but haven't tested this on other operating systems.

If the extension detects that watchman is not installed it will show this popup:
watchman error
With a link to the [install instructions](https://facebook.github.io/watchman/docs/install.html)

My prettier setup was eager to format and sort the installs, I've left it in there. But if you prefer not to have that I can revert those.

Fixes: #22

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -1,142 +1,138 @@
...
 } from "./graphqlUtilsNoVscode";
+import {
+  cacheControl,
+  getCurrentWorkspaceRoot,
+  isReScriptRelayProject,
+  loadFullSchema,
+  loadGraphQLConfig,
+  loadRelayConfig,
+} from "./loadSchema";
+import { featureEnabled, getPreferredFragmentPropName } from "./utils";
```
Comment (zth) [inline src/extension.ts]: Could you please revert this reordering of the imports?

File: src/extension.ts [hunk 2]
```diff
@@ -2304,6 +2300,27 @@ export async function activate(context: ExtensionContext) {
       item.tooltip = "Click to see full output";
     }
 
+    function checkThatWatchmanIsInstalled() {
+      const client = new watchman.Client();
+      client.capabilityCheck({ optional: [], required: [] }, () => {});
+      const installText = "Open install instructions";
+      client.on("error", () =>
+        window
+          .showErrorMessage("Error watchman not installed", installText)
```
Comment (zth) [inline src/extension.ts]: Maybe make this a warning instead, and change the text to reflect that it's needed to be able to run the compiler automatically from the extension?

**Review by zth — CHANGES_REQUESTED** [review]
Great stuff, thank you! 😄  Just 2 minor things from my side.

**Comment by mellson** [comment]
> Great stuff, thank you! 😄 Just 2 minor things from my side.

Thanks, I've updated with the requested changes 👍🏻

**Review by zth — APPROVED** [review]
👏

**Pull request merged**